getColorsFromPairNumber.c

```diff
@@ -0,0 +1,40 @@
+#include<stdio.h>
+#include <assert.h>
+#include"common.h"
+
+static const int MAX_COLORPAIR_NAME_CHARS = 16;
+static const char* MajorColorNames[] = {
+    "White", "Red", "Black", "Yellow", "Violet"
+};
+static const char* MinorColorNames[] = {
+    "Blue", "Orange", "Green", "Brown", "Slate"
+};
+
+void ColorPairToString(const ColorPair* colorPair, char* buffer) {
+    sprintf(buffer, "%s %s",
+        MajorColorNames[colorPair->majorColor],
+        MinorColorNames[colorPair->minorColor]);
+}
+
+ColorPair GetColorFromPairNumber(int pairNumber) {
+    ColorPair colorPair;
+    int numberOfMinorColors = sizeof(MinorColorNames) / sizeof(MinorColorNames[0]);
+    int zeroBasedPairNumber = pairNumber - 1;
+    colorPair.majorColor = 
+        (enum MajorColor)(zeroBasedPairNumber / numberOfMinorColors);
+    colorPair.minorColor =
+        (enum MinorColor)(zeroBasedPairNumber % numberOfMinorColors);
+    return colorPair;
+}
+
+void testNumberToPair(int pairNumber,
```

Comment:
The test code and functionality are in the same file. This means re-using the functionality forces me to take the test code as well. Separate test code from production code.

Reply:
Sure thing, I will go ahead and separate the test portions to a new fie